frontend/model/notifications/vuexModule.js

```diff
@@ -76,11 +76,9 @@ const mutations = {
     if (state.includes(notification)) {
       throw new Error('This notification is already in the store.')
     }
-    if (state[0] && age(notification) > age(state[0])) {
-      throw new Error('This notification is older than the latest one in the store.')
-    }
-    // Using `unshift()` here to insert the new item before older ones.
-    state.unshift(notification)
+    state.push(notification)
+    // Sort items in chronological order, newest items first.
+    state.sort(compareOnAge)
```

Comment:
I'm looking at the implementation of `compareOnAge`, and its usage of `age` doesn't make much sense to me:

```js
// Compares notifications, so that newer ones come first.
export function compareOnAge (notificationA: Notification, notificationB: Notification): -1 | 1 {
  return age(notificationA) <= age(notificationB) ? -1 : 1
}
```

This really should be implemented like so:

```js
export function compareOnAge (notificationA: Notification, notificationB: Notification): number {
  return notificationA.timestamp - notificationB.timestamp

  // OR:
  // return age(notificationA) - age(notificationB)
}
```

Or:

And similarly, `age` doesn't make sense. Instead of:

```js
export function age (notification: Notification): number {
  return Date.now() - notification.timestamp
}
```

It should simply return `notification.timestamp`.

Reply:
@taoeffect Ok, may I rename `compareOnAge` to `compareOnTimestamp` then?

Reply:
Yep!

Reply:
@taoeffect However removing the age function is going to make the `isExpired()` function and the conditions in `compareOnPriority()` a bit less readable